pkg/capacity/capacity.go

```diff
@@ -59,7 +60,103 @@ func FetchAndPrint(showContainers, showPods, showUtil, showPodCount, excludeTain
 	printList(&cm, showContainers, showPods, showUtil, showPodCount, showNamespace, output, sortBy, availableFormat)
 }
 
-func getPodsAndNodes(clientset kubernetes.Interface, excludeTainted bool, podLabels, nodeLabels, namespaceLabels, namespace string) (*corev1.PodList, *corev1.NodeList) {
+// Taints can have two possible formats, key1:effect or key1=value1:effect. This function handles both formats and returns each part as a separate string for comparison.
+func splitTaint(taint string) (string, string, string) {
```

Comment:
Can you use one of the the k8s helper functions instead here? Maybe something like this https://github.com/kubernetes/kubernetes/blob/a07b1aaa5b39b351ec8586de800baa5715304a3f/pkg/apis/core/helper/helpers.go#L428 or https://github.com/kubernetes/kubernetes/blob/a07b1aaa5b39b351ec8586de800baa5715304a3f/pkg/util/taints/taints.go#L91. I know those might not be quite a perfect fit, but maybe there's something along those lines that could help. Also in case it's ever helpful [cs.k8s.io](https://cs.k8s.io) is a k8s-wide codesearch that helps find if helper functions like this already exist - that's what I used here fwiw.

Reply:
ooh cs.k8s.io is nifty! I'll have to do some digging there. 

Apparently you can create node taints that have no value. Like `kubectl taint nodes foo bar:NoSchedule`
as opposed to a k/v taint like `kubectl taint nodes foo dedicated=special-user:NoSchedule`. 
That's what this function is handling. Those 2 possible taint expressions.

I'll look around and see what's been built.

Reply:
I think you're right [`ParseTaints()` helper function](https://github.com/kubernetes/kubernetes/blob/a07b1aaa5b39b351ec8586de800baa5715304a3f/pkg/util/taints/taints.go#L91) can be a replacement for splitTaint(), it will change the user syntax from `kube-capacity --node-taints node.kubernetes.io/disk-pressure,!node.kubernetes.io/unschedulable:NoSchedule` to a more familiar syntax `kube-capacity --node-taints node.kubernetes.io/disk-pressure,node.kubernetes.io/unschedulable:NoSchedule-` with the `-` denoting removing a taint instead of `!`.

I think that's a valuable addition I'll try to get this added in the next few days.

Reply:
splitTaint() has been replaced with k8s helper function ParseTaint(). See 903bc98321dc477a5961f2e2c3f275d49a3e3770